src/server/routers/_app.ts

```diff
@@ -398,6 +398,65 @@ export const appRouter = router({
       return res.data
     }),
 
+  declineProposalApplication: authedProcedure
+    .input(
+      z.object({
+        proposalId: z.string(),
+        proposalApplicationId: z.string(),
+        applicantEmail: z.email(),
+      })
+    )
+    .output(z.object({ success: z.boolean() }))
+    .mutation(async ({ ctx, input }) => {
+      // Check if the user is authorized to decline this application
+      const proposal = await prisma.proposal.findUnique({
+        where: { id: input.proposalId },
+        include: {
+          supervisedBy: {
+            include: {
+              supervisor: true,
+              responsible: true
+            }
+          },
+        },
+      })
+
+      if (!proposal) {
+        throw new TRPCError({
+          code: 'NOT_FOUND',
+          message: 'Proposal not found',
+        })
+      }
+
+      // Verify the user is either a supervisor or a responsible person for this proposal
+      const isAuthorized = proposal.supervisedBy.some(
+        supervision => 
+          supervision.supervisorEmail === ctx.user.email || 
+          supervision.responsible?.email === ctx.user.email
+      )
+
+      if (!isAuthorized) {
+        throw new TRPCError({
+          code: 'FORBIDDEN',
+          message: 'You are not authorized to decline applications for this proposal',
+        })
+      }
+
+      // Update the application status
+      await prisma.proposalApplication.update({
```

Comment:
we also need to add the email sending functionality before we merge this, otherwise we would not notify the student at all